x/nexus/types/types.go

```diff
@@ -144,3 +144,33 @@ func (m MaintainerState) CountIncorrectVotes(window int) uint64 {
 
 // GetAddress returns the address of the maintainer
 func (m MaintainerState) GetAddress() sdk.ValAddress { return m.Address }
+
+// ValidateBasic returns an error if the type is invalid
+func (m RateLimit) ValidateBasic() error {
+	if err := m.Chain.Validate(); err != nil {
+		return err
+	}
+
+	if err := m.Limit.Validate(); err != nil {
+		return err
+	}
+
+	if m.Window.Nanoseconds() <= 0 {
```

Comment:
is that possible?

Reply:
It's a duration type, so it can be negative when subtracting newer timestamp from an older one.